server/services/gameLogic.js

```diff
@@ -131,6 +162,8 @@ exports.playCard = (username, cardId) => {
             cardId: cardId
         };
         cards.push(card);
+        let playerCards =  players.find(player => player.username === username).cards;    
+        playerCards.find(playedCard => playedCard.id.toString() === cardId).played = true;
```

Comment:
Same as above

Reply:
Why not?

Reply:
Because you're making a copy of a list and then changing something in it right? Either way, I don't think we need a 'played' attribute, as we have a list of played cards, so `.played` just becomes `cards.includes`

Reply:
Just to clarify, JavaScript is pass by value but this doesn't hold where object are concerned. Unless you're doing object assign, spread, etc. passing an object is technically pass by reference. So in this case, the find will return an object which is a reference to the one in the array.

Reply:
@tillywoodfield I thought you told me to put a played attribute in the player cards. It's needed because whenever a player refreshes a page, you don't want to return their original hand, you only want to return the ones that they haven't played yet.

Reply:
Ah okay I didn't know that, thanks! I still think we don't need this in the first place though